client/lib/user/user.d.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * External Dependencies
+ */
+import { EventEmitter } from 'events';
+
+export interface User extends EventEmitter {
+	initialize: () => Promise< void >;
+	clearStoreIfChanged: () => void;
+	get: () => UserData;
+	fetch: () => Promise< void > | boolean;
+	handleFetchFailure: ( error: Error ) => void;
+	handleFetchSuccess: ( userdata: UserData ) => void;
+	getLanguage: () => any | undefined;
+	getAvatarUrl: ( options: any ) => string;
+	clear: () => Promise< void >;
+	sendVerificationEmail: requestHandler;
+	set: ( attributes: any ) => boolean;
+	decrementSiteCount: () => void;
+	incrementSiteCount: () => boolean | void;
+	verificationPollerCallback: ( signal: unknown ) => void;
+	checkVerification: () => void;
+	signalVerification: () => void;
+	dispatchToken: string;
+}
+
+/**
+ * Takes either an optional callback function. If no callback is passed in, it will return a promise.
+ * Todo, once types are added for wpcom, move this definition there.
+ * packages/wpcom.js/src/lib/util/send-request.js:83
+ **/
+interface requestHandler {
```

Comment:
Note that we usually capitalize type names: `interface RequestHandler`

Reply:
yup, the linter caught that too